kratos/geometries/line_2d_2.h

```diff
@@ -890,6 +890,57 @@ class Line2D2 : public Geometry<TPointType>
         return false;
     }
 
+    /** Test the intersection with another geometry
+     *  Test if this geometry intersects with other line_2d_2
+     *
+     * @param  rOtherGeometry Geometry to intersect with
+     * @return True if the geometries intersect, False in any other case.
+     */
+    virtual bool HasIntersection(const BaseType& rOtherGeometry) override
```

Comment:
please remove the virtual, it is not needed and afaik gives warnings in some compilers